test_contextual_lenses/tape_gfp.py

```diff
@@ -0,0 +1,673 @@
+"""Contextual lenses on TAPE fluoresence dataset."""
+
+
+import jax.numpy as jnp
+
+import numpy as np
+
+import tensorflow as tf
+
+import numpy as np
+
+import pandas as pd
+
+import matplotlib.pyplot as plt
+
+import scipy.stats
+
+import sklearn.linear_model
+import sklearn.metrics
+
+import seaborn as sns
+
+from contextual_lenses import mean_pool, max_pool, \
+linear_max_pool, linear_mean_pool, gated_conv 
+
+from train_utils import create_optimizer, train_step, \
+train, create_representation_model
+
+from encoders import one_hot_encoder, cnn_one_hot_encoder, \
+one_hot_pos_emb_encoder, cnn_one_hot_pos_emb_encoder
+
+from tape.tape.datasets import LMDBDataset
+
+
+"""Data processing."""
+def dataset_to_df(in_name):
+  dataset = LMDBDataset(in_name)
+  df = pd.DataFrame(list(dataset)[:])
+  df['log_fluorescence'] = df.log_fluorescence.apply(lambda x: x[0])
+  df['binary_log_f'] = [int(b) for b in df['log_fluorescence'] > 2.5]
+  return df
+
+
+"""Padding."""
+SEQ_LEN = 237
+def pad_seq(seq, pad_char='-'):
+  
+  padded_seq = seq + pad_char*(SEQ_LEN-len(seq))
+  padding_mask = []
+  for _ in range(len(seq)):
+    padding_mask.append([1])
+  for _ in range(SEQ_LEN-len(seq)):
+    padding_mask.append([0])
+  padding_mask = np.array(padding_mask)
+
+  return padded_seq, padding_mask
+
+
+"""Open train data and add one-hots."""
+train_df = dataset_to_df('tape/fluorescence/fluorescence_train.lmdb')
+train_df.primary.apply(len).describe()
+
+train_df['padded_primary'] = train_df.primary.apply(lambda x: pad_seq(x)[0])
+train_df['padding_masks'] = train_df.primary.apply(lambda x: pad_seq(x)[1])
+train_df.padded_primary.apply(len).describe()
+
+AMINO_ACID_VOCABULARY = [
+    'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R',
+    'S', 'T', 'V', 'W', 'Y', '-'
+]
+
+
+def residues_to_one_hot_inds(amino_acid_residues):
+  return np.array([AMINO_ACID_VOCABULARY.index(char) for char in amino_acid_residues])
+
+train_df['one_hot_inds'] = train_df.padded_primary.apply(lambda x: residues_to_one_hot_inds(x))
+
+print(train_df.head())
+
+
+"""Open test data and add one-hots."""
+test_df = dataset_to_df('tape/fluorescence/fluorescence_test.lmdb')
+test_df.primary.apply(len).describe()
+
+test_df['padded_primary'] = test_df.primary.apply(lambda x: pad_seq(x)[0])
+test_df['padding_masks'] = test_df.primary.apply(lambda x: pad_seq(x)[1])
+test_df.padded_primary.apply(len).describe()
+
+test_df['one_hot_inds'] = test_df.padded_primary.apply(lambda x: residues_to_one_hot_inds(x))
+
+print(test_df.head())
+
+
+"""Data batching."""
+def create_train_data(epochs, train_bs=256):
```

Comment:
you should avoid closing over global variables as much as possible. Here, you're closing over the data df. I'd have that be an argument to the function.